FaderPlugin/Plugin.cs

```diff
@@ -168,148 +188,338 @@ private void OnChatMessage(XivChatType type, int _, ref SeString sender, ref SeS
+        var target = TargetManager.Target;
+        UpdateState(State.EnemyTarget, target?.ObjectKind == ObjectKind.BattleNpc);
+        UpdateState(State.PlayerTarget, target?.ObjectKind == ObjectKind.Player);
+        UpdateState(State.NPCTarget, target?.ObjectKind == ObjectKind.EventNpc);
+        UpdateState(State.Crafting, Condition[ConditionFlag.Crafting]);
+        UpdateState(State.Gathering, Condition[ConditionFlag.Gathering]);
+        UpdateState(State.Mounted, Condition[ConditionFlag.Mounted] || Condition[ConditionFlag.Mounted2]);
+
+        bool inIslandSanctuary = _territorySheet.HasRow(ClientState.TerritoryType)
+            && _territorySheet.GetRow(ClientState.TerritoryType).TerritoryIntendedUse.RowId == 49;
+        UpdateState(State.IslandSanctuary, inIslandSanctuary);
+
+        var boundByDuty = Condition[ConditionFlag.BoundByDuty]
+                          || Condition[ConditionFlag.BoundByDuty56]
+                          || Condition[ConditionFlag.BoundByDuty95];
+        UpdateState(State.Duty, !inIslandSanctuary && boundByDuty);
+
+        var occupied = Condition[ConditionFlag.Occupied]
+          || Condition[ConditionFlag.Occupied30]
+          || Condition[ConditionFlag.Occupied33]
+          || Condition[ConditionFlag.Occupied38]
+          || Condition[ConditionFlag.Occupied39]
+          || Condition[ConditionFlag.OccupiedInCutSceneEvent]
+          || Condition[ConditionFlag.OccupiedInEvent]
+          || Condition[ConditionFlag.OccupiedSummoningBell]
+          || Condition[ConditionFlag.OccupiedInQuestEvent];
+
+        UpdateState(State.Occupied, occupied);
+    }
 
-        // Chat Focus
-        UpdateStateMap(State.ChatFocus, Addon.IsChatFocused());
+    /// <summary>
+    /// Collects all addon hover states
+    /// </summary>
+    private void UpdateHoverStates()
+    {
+        var mousePos = ImGui.GetMousePos();
+        _addonHoverStates.Clear();
 
-        // Chat Activity
-        UpdateStateMap(State.ChatActivity, HasChatActivity);
+        foreach (var addonName in _addonNameToElement.Keys)
+        {
+            // Compute the hover state once per addon.
+            _addonHoverStates[addonName] = IsAddonHovered(addonName, mousePos);
+        }
+    }
 
-        // Combat
-        UpdateStateMap(State.IsMoving, Addon.IsMoving());
 
-        // Combat
-        UpdateStateMap(State.Combat, Condition[ConditionFlag.InCombat]);
+    private void UpdateMouseHoverState()
+    {
+        // Update the hover states for all addons.
+        UpdateHoverStates();
 
-        // Weapon Unsheathed
-        UpdateStateMap(State.WeaponUnsheathed, Addon.IsWeaponUnsheathed());
+        var currentHovered = new HashSet<string>(
+            _addonHoverStates.Where(kvp => kvp.Value).Select(kvp => kvp.Key)
+        );
 
-        // In Sanctuary (e.g Cities, Aetheryte Villages)
-        UpdateStateMap(State.InSanctuary, Addon.InSanctuary());
+        if (!currentHovered.SetEquals(_previousHoveredAddons))
```

Comment:
Can use SequenceEquals as the sequence should always be the same if nothing changed.
Also remove scoping.